### PR TITLE
[FIX] purchase_stock: non-admin can't receive picking

### LINKED ISSUE #57780: Purchase & Inventory Administrator cannot use the button "Receive Products": Access Error on 'Action Window'
Impacted versions:

saas-13-5

Steps to reproduce:

- Create an internal user who is administrator for both Purchase and Inventory
- Login with that user
- Open a Purchase order
- Click on the button "Receive Products"

Current behavior:

```
Access Error
×
You are not allowed to access 'Action Window' (ir.actions.act_window) records.

This operation is allowed for the following groups:
	- Administration/Settings

Contact your administrator to request access if necessary.
```

Expected behavior:

Video/Screenshot link (optional):
image

### PR DESCRIPTION
**Pull request by yhu-odoo**
Fix #57780

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/purchase_stock/models/purchase.py [hunk 1]
```diff
@@ -126,7 +126,7 @@ def button_cancel(self):
     def action_view_picking(self):
         """ This function returns an action that display existing picking orders of given purchase order ids. When only one found, show the picking immediately.
         """
-        action = self.env.ref('stock.action_picking_tree_all')
+        action = self.env.ref('stock.action_picking_tree_all').sudo()
```
Comment (Whenrow) [inline addons/purchase_stock/models/purchase.py:129]: `sudo()` works but is a little bit too overkill and not so robust against vulnerabilities. Instead, it's safer to get the action values with `_from_xml_id`.
Here is an example : 
https://github.com/odoo/odoo/blob/d3bad3316bcde90ff182decb33c9729c54f9f601/addons/stock/models/product.py#L417-L420

**Review by Whenrow — APPROVED** [review]
LGTM (cc @amoyaux )

**Comment by yhu-odoo** [comment]
@Whenrow also fixed other similar issues in our code base, please check

**Review by Whenrow — APPROVED** [review]

**Comment by amoyaux** [comment]
robodoo r+

**Comment by matteoopenf** [comment]
I have similar errore in v 14.0 this fix have a forward?

**Pull request closed without merging**